flixel/input/gamepad/ButtonIndex.hx

```diff
@@ -0,0 +1,405 @@
+package flixel.input.gamepad;
+import flixel.input.gamepad.ButtonID;
+import flixel.input.gamepad.FlxGamepad.FlxGamepadAnalogStick;
+import flixel.input.gamepad.FlxGamepad.GamepadModel;
+
+/**
+ * ...
+ * @author larsiusprime
+ */
+class ButtonIndex
+{
+	public var model:GamepadModel;
```

Comment:
Oh wait, you're right this is the model variable in ButtonIndex and not Gamepad! This should not be publicly settable. It's Gamepad's that can be settable, I gotta check that logic real quick.